src/tools/imgtool/charconv.cpp

```diff
@@ -2,161 +2,119 @@
 // copyright-holders:Nathan Woods
 /***************************************************************************
 
-    charconv.c
+    charconv.cpp
 
     Imgtool character set conversion routines.
 
 ***************************************************************************/
 
 #include "corestr.h"
 #include "charconv.h"
+#include "unicode.h"
+#include "coretmpl.h"
 
-
-/*-------------------------------------------------
-    utf8_from_latin1 - convert an ISO-8859-1
-    character sequence to an UTF-8 string
--------------------------------------------------*/
-
-static char *utf8_from_latin1(const char *src)
+static const char32_t iso_8859_1_code_page[128] =
+{
+	// 0x80 - 0x8F
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
```

Comment:
The 0x80-0x9f range in ISO-8859-1 is the C1 control block, which exists in the same space in Unicode - it isn't unmapped space.